README.md

```diff
@@ -88,22 +98,50 @@ export interface ErrorMessages {
   callParenthesesMissing: string;
   /** This variable is not defined. */
   undefinedVariable: string;
-  /** Expected at least one parameter in a function call. */
-  expectedAtLeastOneParameter: string;
-  /** Expected that many parameters, found this many. */
-  parameterCountMismatch: (thatMany: number, thisMany: number) => string;
+  /** Expected exactly that many parameters, found this many. */
+  expectedExactArgumentCount: (count: number) => string;
+  /** Expected at least that many parameters, found this many. */
+  expectedMinimumArgumentCount: (count: number) => string;
+  /** Expected between that and that many parameters, found this many. */
+  expectedArgumentCountRange: (
+    actualCount: number,
+    minCount: number,
+    maxCount: number
+  ) => string;
 }
 ```
 
+### Custom Functions
+
+The configurations of custom functions allows defining the minimum and maximum number of arguments that can be passed to each function. When a function supports an unknown number of maximum arguments, `Infinity` should be provided. If `null` is provided as value, by default, the minimum will be `0` and the maximum `Infinity`. When a is provided, both `minArgs` and `maxArgs` are required.
+
+```ts
+/** Configurations functions  */
+export type DentakuFunctionConfig = {
+  minArgs: number;
+  maxArgs: number;
+};
```

Comment:
issue: the name _function config_ sounds a bit generic to me, and hence requires an explanation in the form of documentation. I'd rather give it a more self-explanatory name such as `Arity` and add `@default` annotations to convey the unbounded cases. In this case, I think we can make `minArgs` and `maxArgs` optional, as it will be obvious which value will be used if either of these fields is not provided.

suggestion:
```suggestion
export type Arity = {
  /** @default 0 */
  minArgs?: number;
  /** @default Infinity */
  maxArgs?: number;
};
```